### ISSUE TITLE
Email address InlineEdit not working (Contact, Account, ...) in SuiteCRM 7.x

### ISSUE THREAD

**Issue by gody01** [salesagility/SuiteCRM#9762]
#### Issue
Inline Edit of Email adress in Contact, Account doesn't save new Email.
It works in Version 7.11.18

On Contact or Account detail view, when You try to change Email address with inline edit, after confirming change, Email changes back to original and change is not saved.

#### Expected Behavior

Changed Email should be saved and displayed.

#### Actual Behavior

Changes are not saved, and old Email address is displayed in UI also.

#### Steps to Reproduce

1. Open Detail View of Contact ot Account with defined Email address
2. Click on Edit icon on Email field
3. Make change to Email Address
4. Click on save icon or press enter
5. Changed Email revert to old Email. On page refresh, there is still old Email too (change not not saved to DB).

Same steps in 7.11.18 normaly changed and saves new Email adress

#### Context

Trying to correct Email address with inline editing.
Has to Edit complete record, to make change to Email adress.

#### Your Environment

* SuiteCRM Version used: 7.12.5 & 7.12.7
* Browser name and version Chrome, Firefox / linux, windows
* Environment name and version  Mysql: mariadb-server-10.3.32, php-7.3.33 (SuiteCRM 7.12.7)
* Environment name and version  Mysql: mariadb-server-10.3.28, php-7.3.33 (SuiteCRM 7.12.5)

* Operating System and version Linux RHEL 8, CentOS 8, RockyLinux 8

**Comment by pgorod** [comment]
Perhaps related to this change?

#9409

Can you please try undoing that change and see if it fixes your bug?

**Comment by gody01** [comment]
Unfortunately not.

Request, which returns old Email is:

"GET /index.php?module=Home&action=saveHTMLField&field=email1&current_module=Contacts&id=7f117037-082a-fad4-66a5-62e8e66af2a2&value=mail.address%40domain.si&view=DetailView&parent_type=&to_pdf=true HTTP/1.0" 200 369 "https://crm.gody.si/index.php?action=ajaxui"

**Comment by gody01** [comment]
```
       if (($bean->ACLAccess("edit") || is_admin($current_user)) && $enabled) {
            $bean->in_workflow=true;
            if ($field == 'email1') {
                $bean->email1_set_in_workflow=true;
            }
            if (!$bean->save($check_notify)) {
                $GLOBALS['log']->fatal("Saving probably failed or bean->save() method did not return with a positive result.");
            }
        } else {
            $GLOBALS['log']->fatal("ACLAccess denied to save this field.");
        }
---> up to here $bean->field is still correct (new)

        $bean->retrieve();

---> here old email is retrieved
```

It looksl. like function $bean->save wouldn't save email1. But it doesn't return false ...

**Comment by pgorod** [comment]
There seems to be some additional logic for this in `include/SugarObjects/templates/company/Company.php` and `include/SugarObjects/templates/person/Personq.php`

I wonder if it makes a difference if you're changing to an email that already exists in EmailAddresses table; or to one that doesn't exist there yet. I know that the **`email1`** field is just short-hand for "the first related email in EmailAddresses table", but keeping those two bits of data in sync sometimes gets complicated...

**Comment by gody01** [comment]
Yes, I compared debug logs from 7.11.18 and 7.12.7 and in 7.11.18 there are queries to:
email_addr_bean_rel
table, while in 7.12.7 there is none.

**Comment by gody01** [comment]
Just tested.
Changing to existing email address also doesn't work.

**Comment by phil8900** [comment]
Could someone take a look at the fix proposed [here](https://community.suitecrm.com/t/suitecrm-7-13-inline-edit-not-working-for-email-in-leads-module/97923)?

It seems to work.

**Comment by phil8900** [comment]
I think I found the root cause: inline edit deliberately sets email1_set_in_workflow=true, but the later save code only copies that boolean flag as an email value, not the edited address.

- If $this->addresses is empty, keep the original full legacy parse from email1..email10.
- If $this->addresses is already populated, merge legacy bean fields into it instead of clearing it.
- The merge replaces the entries represented by populateLegacyFields() as email1/email2, so inline edits update/delete those addresses.
- Addresses not representable by populateLegacyFields() (email3+) remain preserved.

We use this right now:
[SugarEmailAddress.php](https://github.com/user-attachments/files/30701363/SugarEmailAddress.php)
